lxd/container_lxc.go

```diff
@@ -1369,8 +1373,26 @@ func (c *containerLXC) initLXC() error {
 	return nil
 }
 
+// Initialize the storage pool for this container.
+func (c *containerLXC) initStoragePool() error {
+	if c.storagePool != "" {
+		return nil
+	}
+
+	_, rootDiskDevice := containerGetRootDiskDevice(c.localDevices)
+	if rootDiskDevice["pool"] == "" {
+		_, rootDiskDevice = containerGetRootDiskDevice(c.expandedDevices)
+	}
+	if rootDiskDevice["pool"] == "" {
+		return fmt.Errorf("No storage pool found in the containers devices.")
+	}
```

Comment:
We can reverse engineer the pool the container is on when it is already created but this function here needs to be called at least in `containerLXCCreate()`.